lib/tasks/npm-task.js

```diff
@@ -110,13 +118,17 @@ class NpmTask extends Task {
     if (this.useYarn === true) {
       logger.info('yarn requested -> trying yarn');
 
-      return this.checkYarn().catch((error) => {
-        if (error.code === 'ENOENT') {
-          throw new SilentError('Yarn could not be found.');
-        }
-
-        throw error;
-      });
+      return this.checkYarn()
+        .catch((error) => {
+          if (error.code === 'ENOENT') {
+            throw new SilentError('Yarn could not be found.');
+          }
+
+          throw error;
+        })
+        .then(({ yarnVersion }) => {
+          this.yarnVersion = yarnVersion;
+        });
```

Comment:
Would you mind tweaking this to use `async`/`await` while you are making changes? Not sure why we missed doing that as part of the other work (adding `async` to `findPackageManager`)...

I'm thinking something like:


```suggestion
      try {
        await this.checkYarn()
      } catch(error) {
        if (error.code === 'ENOENT') {
          throw new SilentError('Yarn could not be found.');
        }

        throw error;
      }
```

Reply:
I spent a while fixing the test stubs because of the refactoring of where the errors are getting thrown due to your other comment, and I think I also removed the `.then()` flow incidentally. Whatever happened, the only results for <kbd>⌘</kbd> + <kbd>F</kbd> `.then()` brings up the lines for the workaround for npm 5 and 6.